frontend/subrequests/lint/lint.go

```diff
@@ -40,8 +40,9 @@ type Warning struct {
 }
 
 type LintResults struct {
-	Warnings []Warning        `json:"warnings"`
-	Sources  []*pb.SourceInfo `json:"sources"`
+	Warnings   []Warning        `json:"warnings"`
+	Sources    []*pb.SourceInfo `json:"sources"`
+	BuildError string           `json:"buildError,omitempty"`
```

Comment:
Make it a typed error with the same source info as `Warning`

Reply:
The most straightforward way to do this that doesn't require quite a bit of life is to modify `toDispatchState` so that it also returns `pb.Location`,

`func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchState, error)`

Is that what you're suggesting?